cloudinit/config/cc_reset_rmc.py

```diff
@@ -0,0 +1,167 @@
+# (c) Copyright IBM Corp. 2020 All Rights Reserved
+#
+# Author: Aman Kumar Sinha <amansi26@in.ibm.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+
+"""
+Reset RMC
+------------
+**Summary:** reset rsct node id
+
+Reset RMC module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components,
+that  together provide a comprehensive clustering environment (RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource monitoring
+and control (RMC) subsystem. RMC is a generalized framework used for managing,
+monitoring, and manipulating resources. RMC runs as a daemon process on
+individual machines and needs creation of unique node id and restarts
+during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- creation of the unique RSCT node id to every instance/virtual machine
+  and ensure once set, it isn't changed subsequently by cloud-init.
+  In order to do so, it restarts RSCT service.
+
+Prerequisite of using this module is to install RSCT packages.
+
+**Internal name:** per instance
+
+**Supported distros:** rhel, sles and ubuntu
+
+"""
+
+import json
+import os
+import time
+
+from cloudinit.settings import PER_INSTANCE
+from cloudinit import util
+from cloudinit import subp
+
+frequency = PER_INSTANCE
+PIDOF = 'pidof'
+RMCCTRL = '/usr/sbin/rsct/bin/rmcctrl'
+RECFGCT = '/usr/sbin/rsct/install/bin/recfgct'
+SRCMSTR = '/sbin/srcmstr'
+
+NODE_ID_FILE = '/etc/ct_node_id'
+
+SRCMSTR_TIMEOUT_SECONDS = 30
+
+
+def handle(name, _cfg, _cloud, log, _args):
+    global RMCCTRL, RECFGCT
+    with open('/run/cloud-init/instance-data.json') as data_file:
+        ds = json.load(data_file)
+        ds_value = ds['v1'].get('platform')
+
+    # Ensuring node id has to be generated only once during first boot
+    if ds_value == 'none':
+        log.debug('Skipping creation of new ct_node_id node')
+        return
+    required_tools = [RMCCTRL, RECFGCT]
+    for tool in required_tools:
```

Comment:
I think this is a little bit more palatable from a code perspective, but should still give you what you were after.
By setting the PATH to prefix with the paths you want, and ensuring that the tools are in one of those PATHs, then we can safely use 'util.subp' elsewhere without having hard coded paths.

my apologies for not testing or vetting flake8 or pylint.

```python
PIDOF = 'pidof'
RMCCTRL = 'rmcctrl'
RECFGCT = 'recfgct'
SRCMSTR = 'rcmstr'

# We want to ensure the programs invoked are from one of
# these paths, which are unlikely to be in system PATH.
RSCT_PATHS = ['/usr/sbin/rsct/bin', '/usr/sbin/rsct/install/bin',
              '/opt/rsct/bin', '/opt/rsct/install/bin', '/sbin']

NODE_ID_FILE = '/etc/ct_node_id'

SRCMSTR_TIMEOUT_SECONDS = 30


def handle(name, _cfg, _cloud, log, _args):
    with open('/run/cloud-init/instance-data.json') as data_file:
        ds = json.load(data_file)
        ds_value = ds['v1'].get('platform')

    # Ensuring node id has to be generated only once during first boot
    if ds_value == 'none':
        log.debug('Skipping creation of new ct_node_id node')
        return

    ppaths = [p for p in RSCT_PATHS if os.path.isdir(p)]
    missing = [t for t in [RMCCTRL, RECFGCT, SRCMSTR ]
               if not subp.which(t, search=ppaths)]

    if len(missing) != 0:
        log.debug("Missing required tools not attempting reset: %s", missing)

    orig_path = os.environ.get('PATH')
    try:
        suff = ":" + orig_path if orig_path else ""
        os.environ['PATH'] = ":".join(ppaths) + suff
        refresh_imc()
    finally:
        if orig_path is None:
            del os.environ['PATH']
        else:
            os.environ['PATH'] = orig_path


def refresh_imc()
   ...
```

Reply:
@smoser Thanks for guiding me on the way to write a much more readable and better way to handle my scenario.
This helps me to understand how to handle the all the file existence check in one go. As this is my first contribution to the community. thanks for an effort to help me to contribute and thanks for all your support.